central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -95,6 +95,8 @@ func (m *managerImpl) ProcessScanRequest(ctx context.Context, scanRequest *stora
 		return nil, errors.Errorf("Compliance is disabled. Cannot process scan request: %q", scanRequest.GetScanConfigName())
 	}
 
+	createScanRequest := false
+
 	// User MUST have permissions on all clusters being applied.
 	clusterScopeKeys := make([][]sac.ScopeKey, 0, len(clusters))
 	for _, cluster := range clusters {
```

Comment:
I am quite new to these codes. It looks like the clusters are cluster ids. What if I have a set of clusters and the last of them does not exist? Are we going to rollback the changes that are made already?

Reply:
So basically you have to have write permissions on the clusters you pass in for a scan configuration.  So the first thing we do is verify you have those permissions.  If you don't have it on one cluster or one of the clusters you pass in is not valid, then the entire request is rejected before we start to perform the work.

Reply:
I meant we have the loop around clusters, and within the loop, we have two per cluster actions
1. SendMessag   
2. UpdateClusterStatus
For exmaple you have 5 clusters [1,2,3,4,5]
If we have the 3rd cluster failed to update cluster status for example, 1,2 has completed the update and 4,5 not started yet and 3 is in the middle. Do we have something to cleanup or it is not needed?

Reply:
Oh I'm following.  If the 3rd one failed to update the cluster we should continue trying to update 4 and 5.  I will make that adjustment.

Reply:
And also, if we have a failure originated from this loop, the scan configuration is already persisted. It looks like we should cleanup on failure.

Reply:
On nvm.  I think it is working OK and I don't think we need to clean that up.  those that worked would be running and I think that is OK.  At least for now.

Reply:
I think this is not a simple as that as would that mean we also have to send deletes to sensor to delete what we just added.  I think I prefer the path of a separate ticket so as to not block this ticket the UI needs.

https://issues.redhat.com/browse/ROX-22010